sql/core/src/test/scala/org/apache/spark/sql/sources/v2/DataSourceV2SQLSuite.scala

```diff
@@ -196,12 +199,10 @@ class DataSourceV2SQLSuite extends QueryTest with SharedSQLContext with BeforeAn
   }
 
   test("CreateTable: use default catalog for v2 sources when default catalog is set") {
-    val sparkSession = spark.newSession()
-    sparkSession.conf.set("spark.sql.catalog.testcat", classOf[TestInMemoryTableCatalog].getName)
-    sparkSession.conf.set("spark.sql.default.catalog", "testcat")
-    sparkSession.sql(s"CREATE TABLE table_name (id bigint, data string) USING foo")
+    spark.conf.set("spark.sql.default.catalog", "testcat")
```

Comment:
`withSQLConf`? Otherwise you affect all tests below

Reply:
This test suite will clear all the configs after each test, see https://github.com/apache/spark/pull/25368/files#diff-b49f76fba19ee10a28e0e61c4b44e1a0R62